src/python/pants/invalidation/cache_manager.py

```diff
@@ -274,6 +259,13 @@ def __init__(self,
     self._artifact_write_callback = artifact_write_callback
     self.invalidation_report = invalidation_report
 
+    # Create the task-versioned prefix of the results dir, and a stable symlink to it (useful when debugging).
+    self._results_dir_prefix = os.path.join(results_dir_root, sha1(self._task_version).hexdigest()[:12])
+    safe_mkdir(self._results_dir_prefix)
+    stable_prefix = os.path.join(results_dir_root, self._STABLE_DIR_NAME)
+    safe_delete(stable_prefix)
```

Comment:
Obviously doesn't hurt, but I think relative_symlink will gracefully repoint the existing symlink without requiring the delete.

We had to more anal about it in create_results_dirs because we explicitly wanted to catch places where the symlink had been replaced by a real dir.